internal/monitors/expvar/expvar.go

```diff
@@ -0,0 +1,393 @@
+		}
+		mConf.metricType = func() datapoint.MetricType {
+			if strings.TrimSpace(strings.ToLower(mConf.Type)) == gauge {
+				return datapoint.Gauge
+			}
+			return datapoint.Counter
+		}()
+		if mConf.DimensionConfigs == nil {
+			mConf.DimensionConfigs = []*DimensionConfig{{}}
+		}
+		mConf.keys = strings.Split(mConf.JSONPath, ".")
+	}
+	// Initializing default metrics
+	for _, path := range memstatsMetricPathsGauge {
+		conf.MetricConfigs = append(conf.MetricConfigs, &MetricConfig{Name: toSnakeCase(path), JSONPath: path, Type: gauge, metricType: datapoint.Gauge, DimensionConfigs: []*DimensionConfig{{}}, keys: strings.Split(path, ".")})
+	}
+	for _, path := range memstatsMetricPathsCounter {
+		conf.MetricConfigs = append(conf.MetricConfigs, &MetricConfig{Name: toSnakeCase(path), JSONPath: path, Type: counter, metricType: datapoint.Counter, DimensionConfigs: []*DimensionConfig{{}}, keys: strings.Split(path, ".")})
+	}
+}
+
+// Configure monitor
+func (m *Monitor) Configure(conf *Config) error {
+	conf.setURL()
+	conf.setRunInterval()
+	conf.initMetrics()
+	m.setContext()
+	m.setClient()
+	utils.RunOnInterval(m.ctx, func() {
+		dpsMap, err := m.fetchMetrics(conf)
+		if err != nil {
+			logger.WithError(err).Error("could not get expvar metrics")
+			return
+		}
+		mostRecentGCPauseIndex := getMostRecentGCPauseIndex(dpsMap)
+		now := time.Now()
+		for metricPath, dps := range *dpsMap {
+			for _, dp := range dps {
+				if err := m.sendDatapoint(conf, dp, metricPath, mostRecentGCPauseIndex, &now); err != nil {
+					logger.Error(err)
+				}
+			}
+		}
+	}, conf.runInterval)
+	return nil
+}
+
+func (m *Monitor) sendDatapoint(conf *Config, dp *datapoint.Datapoint, metricPath string, mostRecentGCPauseIndex int64, now *time.Time) (err error) {
+	if metricPath == memstatsPauseNsMetricPath || metricPath == memstatsPauseEndMetricPath {
+		var index int64
+		if index, err = strconv.ParseInt(dp.Dimensions[metricPath], 10, 0); err == nil && index == mostRecentGCPauseIndex {
+			dp.Metric = memstatsMostRecentGCPauseNsMetricName
+			if metricPath == memstatsPauseEndMetricPath {
+				dp.Metric = memstatsMostRecentGCPauseEndMetricName
+			}
+			// For index dimension key is equal to metric path for default metrics memstats.PauseNs and memstats.PauseEnd
+			delete(dp.Dimensions, metricPath)
+		} else {
+			if err != nil {
+				err = fmt.Errorf("failed to set metric most recent GC pause metric. %+v", err)
+			}
+			return err
+		}
+	}
+	// Renaming auto created dimension 'memstats.BySize' to 'class'
+	if metricPath == memstatsBySizeSizeMetricPath || metricPath == memstatsBySizeMallocsMetricPath || metricPath == memstatsBySizeFreesMetricPath {
+		dp.Dimensions["class"] = dp.Dimensions[memstatsBySizeDimensionPath]
+		delete(dp.Dimensions, memstatsBySizeDimensionPath)
+	}
+	dp.Timestamp = *now
+	dp.Dimensions["application_host"] = conf.Host
+	dp.Dimensions["application_expvar_port"] = strconv.FormatInt(int64(conf.Port), 10)
```

Comment:
Is this really useful for charts and analytics?  If people really wanted to distinguish ports, they could do it via `extraDimensions`, either directly in the config or with container labels.